yogstation/code/modules/mob/living/simple_animal/borer/borer_event.dm

```diff
@@ -0,0 +1,65 @@
+/datum/round_event_control/borer
+	name = "Borer"
+	typepath = /datum/round_event/borer
+	weight = 15
+	max_occurrences = 1
+	min_players = 20
+
+	earliest_start = 12000
+
+/datum/round_event/borer
+	announceWhen = 3000 //Borers get 5 minutes till the crew tries to murder them.
+	var/spawned = 0
+
+	var/spawncount
+
+/datum/round_event/borer/announce()
+	if(spawned)
+		priority_announce("Unidentified lifesigns detected coming aboard [station_name()]. Secure any exterior access, including ducting and ventilation.", "Lifesign Alert", 'sound/AI/aliens.ogg') //Borers seem like normal xenomorphs.
+
+
+/datum/round_event/borer/start()
+	var/list/vents = list()
+	for(var/obj/machinery/atmospherics/components/unary/vent_pump/temp_vent in GLOB.machines)
+		if(QDELETED(temp_vent))
+			continue
+
+		if(is_station_level(temp_vent.loc.z) && !temp_vent.welded)
+			var/datum/pipeline/temp_vent_parent = temp_vent.parents[1]
+			if(temp_vent_parent.other_atmosmch.len > 20)
+				vents += temp_vent
+
+	if(!vents.len)
+		return kill()
+
+
+	var/total_humans = 0
+	for(var/mob/living/carbon/human/H in GLOB.alive_mob_list)
+		if(H.stat != DEAD)
+			total_humans++
+
+	var/list/mob/dead/observer/candidates = pollCandidates("Do you want to play as a borer?", ROLE_BORER, null, FALSE, 200)
+	if(!candidates.len)
+		return kill()
+
+	for(var/client/C in candidates)
+		if(!(C.prefs.toggles & MIDROUND_ANTAG))
```

Comment:
doesn't this mean they're first asked whether they wanna be a borer, then don't get it if they have the midround antag preference turned off?

Reply:
it should work as intended if pollcandidates wasn't changed to ignore midround antag